rust/src/delta.rs

```diff
@@ -613,6 +673,127 @@ impl std::fmt::Debug for DeltaTable {
     }
 }
 
+#[derive(thiserror::Error, Debug)]
+pub enum DeltaTransactionError {
+    #[error("Version already exists")]
+    VersionExists,
+
+    #[error("Maximum number of retries exceeded for committing the transaction")]
+    CommitRetriesExceeded,
+
+    #[error("RecordBatch is missing partition column in Delta schema.")]
+    MissingPartitionColumn,
+
+    #[error("Commit failed: {source}")]
+    Storage { source: StorageError },
+
+    #[error("DeltaTable interaction failed: {source}")]
+    DeltaTable { source: DeltaTableError },
+}
+
+impl From<StorageError> for DeltaTransactionError {
+    fn from(error: StorageError) -> Self {
+        match error {
+            StorageError::AlreadyExists => DeltaTransactionError::VersionExists,
+            _ => DeltaTransactionError::Storage { source: error },
+        }
+    }
+}
+
+impl From<DeltaTableError> for DeltaTransactionError {
+    fn from(error: DeltaTableError) -> Self {
+        DeltaTransactionError::DeltaTable { source: error }
+    }
+}
```

Comment:
you can avoid this kind of pass through wrapper code by using `from` attribute defined in thiserror:

```rust
    #[error("DeltaTable interaction failed: {source}")]
    DeltaTable {
        #[from]
        source: DeltaTableError,
    },
```

Reply:
neat. TIL... fixed.